helix-term/src/commands.rs

```diff
@@ -4010,26 +4010,15 @@ pub mod insert {
         {
             // TODO: what if trigger is multiple chars long
             let is_trigger = triggers.iter().any(|trigger| trigger.contains(ch));
+            // lsp doesn't tell us when to close the signature help, so we request
+            // the help information again after common close triggers which should
+            // return None, which in turn closes the popup.
+            let close_triggers = &[')', ';'];
```

Comment:
How does this deal with nested signature help?

```
foo(bar(|),  ... )
```

You might need to search for the matching opening brace (and track starting trigger) to determine if the closing brace matches the whole function call

Reply:
The popup is not _forcefully_ closed on `close_triggers`, it simply sends a normal signature help request. If the response from the server is None, we close the popup in `signature_help()`. So as long as the LSP responds correctly to the cursor position, we won't have to do any manual bookkeeping.